src/main/webapp/app/entities/participation/participation.model.ts

```diff
@@ -54,7 +54,7 @@ export abstract class Participation implements BaseEntity {
  *
  * This is needed as different participation types may use different exercise attributes (e.g. not 'exercise' but 'programmingExercise').
  */
-export const getExercise = (participation: Participation | undefined): Exercise | undefined => {
+export const getExercise = (participation: Participation): Exercise | undefined => {
     if (participation) {
```

Comment:
You forgot to remove the if clause here 😅

Reply:
Sadly, this is needed, as the client tests otherwise fail. There seems to be some code relying on the existence of this `if`.

Reply:
What client tests fail? Because how can participation be here not truthy? I'm confused there :D

Reply:
Also, see aa8379e1124c922509bca12aa44d0bdcc67d1d0f and https://github.com/ls1intum/Artemis/runs/3599962633#step:5:1488. While this is one error, it is the only one the tests found, and now I am not sure how much I can trust the type checking / our code base here.

Reply:
I know what the issue is: In `result.component.ts` `getExercise()` gets called if the participation was not set.
participation is a non optional Participation and therefore the typecheck resolves. But in the HTML apparently no participation gets set => getExercise gets called with undefined.

So you could add a undefined check in the `evaluateTemplateStatus()`. Then removing the if clause should work.

Reply:
Looking through the code, there are more occurrences that would need to be checked and more usages of such "unnecessary" if's. In short, I don't know where we need to add if's to remove this single one. I would suggest letting it be for now because it works, and I don't know whether my refactored version will do so.

Reply:
ok